source/beamline/SGM/SGMBeamCoordinatorControl.cpp

```diff
@@ -0,0 +1,252 @@
+	controlsToOpen_.append(vvr1611_3_I10_03Valve_);
+	controlsToOpen_.append(vvr1611_3_I10_04Valve_);
+	controlsToOpen_.append(vvr1611_4_I10_01Valve_);
+	controlsToOpen_.append(vvr1611_4_I10_02Valve_);
+	controlsToOpen_.append(vvr1611_4_I10_03Valve_);
+	controlsToOpen_.append(vvr1611_4_I10_04Valve_);
+	controlsToOpen_.append(vvr1611_4_I10_05Valve_);
+	controlsToOpen_.append(vvr1611_4_I10_06Valve_);
+	controlsToOpen_.append(vvr1611_4_I10_07Valve_);
+	controlsToOpen_.append(vvr1611_4_I10_08Valve_);
+	controlsToOpen_.append(frontBypassValve_);
+	controlsToOpen_.append(backBypassValve_);
+	controlsToOpen_.append(vvr1611_4_I10_11Valve_);
+	controlsToOpen_.append(vvr1611_3_I00_01Valve_);
+	controlsToOpen_.append(vvr1611_3_I00_02Valve_);
+
+
+	controlsToWait_.append(psh1411_I00_01Shutter_);
+	controlsToWait_.append(vvr1411_I00_01Valve_);
+	controlsToWait_.append(vvf1411_I00_01_);
+
+	setMinimumValue(0);
+	setMaximumValue(1);
+	setTolerance(0.5);
+
+	setAllowsMovesWhileMoving(false);
+	setAttemptMoveWhenWithinTolerance(false);
+	updateStates();
+}
+
+bool SGMBeamCoordinatorControl::canMeasure() const
+{
+	bool canMeasureResult = true;
+	foreach(AMControl* childControl, children_) {
+		canMeasureResult &= childControl->canMeasure();
+	}
+
+	return canMeasureResult;
+}
+
+bool SGMBeamCoordinatorControl::canMove() const
+{
+	return isConnected();
+}
+
+bool SGMBeamCoordinatorControl::canStop() const
+{
+	return false;
+}
+
+void SGMBeamCoordinatorControl::updateValue()
+{
+	if(moveInProgress()) {
+		return;
+	}
+
+	if(fastShutterVoltage_->withinTolerance(5)) {
+
+		setValue(0);
+	} else {
+
+		bool isOpen = true;
+
+		if(es2Bypass_->withinTolerance(1)) {
+			isOpen &= frontBypassValve_->withinTolerance(1);
+			isOpen &= backBypassValve_->withinTolerance(1);
+		}
+
+		isOpen &= vvr1611_3_I10_01Valve_->withinTolerance(1);
+		isOpen &= vvr1611_3_I10_02Valve_->withinTolerance(1);
+		isOpen &= vvr1611_3_I10_03Valve_->withinTolerance(1);
+		isOpen &= vvr1611_3_I10_04Valve_->withinTolerance(1);
+		isOpen &= psh1611_3_I10_01Shutter_->withinTolerance(1);
+		isOpen &= vvr1611_4_I10_01Valve_->withinTolerance(1);
+		isOpen &= vvr1611_4_I10_02Valve_->withinTolerance(1);
+		isOpen &= vvr1611_4_I10_03Valve_->withinTolerance(1);
+		isOpen &= vvr1611_4_I10_04Valve_->withinTolerance(1);
+		isOpen &= vvr1611_4_I10_05Valve_->withinTolerance(1);
+		isOpen &= vvr1611_4_I10_06Valve_->withinTolerance(1);
+		isOpen &= vvr1611_4_I10_07Valve_->withinTolerance(1);
+		isOpen &= vvr1611_4_I10_08Valve_->withinTolerance(1);
+		isOpen &= vvr1611_4_I10_11Valve_->withinTolerance(1);
+		isOpen &= vvr1611_3_I00_01Valve_->withinTolerance(1);
+		isOpen &= vvr1611_3_I00_02Valve_->withinTolerance(1);
+
+		isOpen &= psh1411_I00_01Shutter_->withinTolerance(1);
+		isOpen &= vvr1411_I00_01Valve_->withinTolerance(1);
+		isOpen &= vvf1411_I00_01_->withinTolerance(1);
+		isOpen &= safetyShutter_->withinTolerance(1);
+		isOpen &= psh_2_->withinTolerance(1);
+
+		if(isOpen) {
```

Comment:
The pseudo motor uses its moving state to signal the 'in-between' state. This is taken care of in the coordinator (when the beam on pseudo motor is moving it sets its beam state PV to moving).